mt_sat/mt_sat_wrapper.m

```diff
@@ -78,12 +78,33 @@ function mt_sat_wrapper(mtw_nii,pdw_nii,t1w_nii,mtw_jsn,pdw_jsn,t1w_jsn,varargin
 % This env var will be consumed by qMRLab
 setenv('ISNEXTFLOW','1');
 
-if nargin >6
-if any(cellfun(@isequal,varargin,repmat({'qmrlab_path'},size(varargin))))
-    idx = find(cellfun(@isequal,varargin,repmat({'qmrlab_path'},size(varargin)))==1);
-    qMRdir = varargin{idx+1};
+p = inputParser();
+
+%Input parameters conditions
+validNii = @(x) exist(x,'file') && strcmp(x(end-5:end),'nii.gz');
+validJsn = @(x) exist(x,'file') && strcmp(x(end-3:end),'json');
+validB1factor = @(x) isnumeric(x) && (x > 0 && x <= 1);
+
+%Add REQUIRED Parameteres
+addRequired(p,'mtw_nii',validNii);
+addRequired(p,'pdw_nii',validNii);
+addRequired(p,'t1w_nii',validNii);
+addRequired(p,'mtw_jsn',validJsn);
+addRequired(p,'pdw_jsn',validJsn);
+addRequired(p,'t1w_jsn',validJsn);
+
+%Add OPTIONAL Parameteres
+addParameter(p,'mask',validNii);
```

Comment:
If you add these using defaults as empty such as:

`addParameter(p,'mask,[],validNii);`

then you can check them with `isempty(p.Results.mask)`  for the rest of the code.

Reply:
Done!